aspnetcore/mvc/advanced/custom-model-binding/sample/CustomModelBindingSample/Binders/AuthorEntityBinder.cs

```diff
@@ -0,0 +1,75 @@
+﻿using CustomModelBindingSample.Data;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CustomModelBindingSample.Binders
+{
+    public class AuthorEntityBinder : IModelBinder
+    {
+        private readonly AppDbContext _db;
+        public AuthorEntityBinder(AppDbContext db)
+        {
+            _db = db;
+        }
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            // specify a default argument name if none is set
+            var modelName = bindingContext.ModelName;
+            if(string.IsNullOrEmpty(modelName))
+            {
+                modelName = "authorId";
+            }
```

Comment:
Model binders don't generally need to do this. What's happening here if you see `string.Empty` is that that name of the parameter/property wasn't found in the input. So this is a fallback and you're doing a second pass. That should be removed or explained because it's not something our built-in model binders do.